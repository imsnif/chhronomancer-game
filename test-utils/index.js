module.exports = {
  mockFB () {
    const login = require('react-native-facebook-login')
    const FBLogin = jest.fn()
    FBLogin.mockReturnValue(null)
    login.FBLogin = FBLogin
    login.FBLoginManager = {LoginBehaviors: {Web: 2}}
    return FBLogin
  },
  navigatorMock (routeName) {
    return {
      getCurrentRoutes: () => [routeName],
      push: () => {}
    }
  }
}
