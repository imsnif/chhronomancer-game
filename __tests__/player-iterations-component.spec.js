import 'react-native'
import React from 'react'
import serverMock from '../server-mock'
import renderer from 'react-test-renderer'
import { navigatorMock } from '../test-utils'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))
jest.useFakeTimers()

beforeEach(() => {
  serverMock()
})

afterEach(() => {
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
})

test('Player Iterations => renders correctly', () => {
  const PlayerIterations = require('../components/player-iterations').default
  const tree = renderer.create(
    <PlayerIterations navigator={navigatorMock()} playerId={1} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
