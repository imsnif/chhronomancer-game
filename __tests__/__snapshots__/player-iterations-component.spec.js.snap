// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Iterations => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#329932",
      "borderColor": "#329932",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 7,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "textAlign": "center",
        }
      }
    >
      Iterations
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#141313",
        "borderColor": "#141313",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 13,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "column",
                "height": 53.36,
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#00ff00",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 15.303777311500582,
                  "marginTop": 5,
                  "paddingRight": 9,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Timeline 1
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "fontSize": 12.243021849200467,
                    "marginLeft": 9,
                    "paddingTop": 2,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  }
                }
              >
                Resetting
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 7,
                    "fontFamily": "telegrama_raw",
                    "fontSize": 12.243021849200467,
                    "paddingTop": 2,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  }
                }
              >
                | |==------------------------------------------------|   1% 00:01:39
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "column",
                "height": 53.36,
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#00ff00",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 15.303777311500582,
                  "marginTop": 5,
                  "paddingRight": 9,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Timeline 2
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "fontSize": 12.243021849200467,
                    "marginLeft": 9,
                    "paddingTop": 2,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  }
                }
              >
                Locking
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 7,
                    "fontFamily": "telegrama_raw",
                    "fontSize": 12.243021849200467,
                    "paddingTop": 2,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  }
                }
              >
                | |==------------------------------------------------|   1% 00:06:39
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
